### PR TITLE
Run all tests on both Java 8 and Java 10

### PR DESCRIPTION
**Pull request by adaszko**
* [JIRA ticket](https://rchain.atlassian.net/browse/CORE-1246)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -78,3 +134,34 @@ deploy:deploy-rnode-artifacts:
...
+    - artifacts_dir=$(mktemp -d /tmp/artifacts_out.XXXXXXXX)
+    - cp node/target/rnode_*_all.deb ${artifacts_dir}/
+    - cp node/target/rpm/RPMS/noarch/rnode-*.noarch.rpm ${artifacts_dir}/
+    - cp rosette/build.out/rosette-*.deb ${artifacts_dir}/
+    - cp node/target/universal/*.tgz ${artifacts_dir}/
+    # deploy artifacts to dispatcher
+    - mkdir -p ~/.ssh
+    - echo "${SSH_PRIVATE_KEY}" | tr -d '\r' > ~/.ssh/id_rsa
+    - chmod 600 ~/.ssh/id_rsa
+    - rsync -avrz --delete -e  "ssh -p 22 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" ${artifacts_dir}/* inbound@dispatch.pyr8.io:/home/inbound/cd/artifacts-java10/${CI_COMMIT_REF_NAME}/
```
Comment (adaszko) [inline .gitlab-ci.yml]: TODO Make sure this dir exists

File: build.sbt [hunk 2]
```diff
@@ -187,7 +187,7 @@ lazy val node = (project in file("node"))
     /* Dockerization */
     dockerUsername := Some(organization.value),
     dockerUpdateLatest := true,
-    dockerBaseImage := "openjdk:8u171-jre-slim-stretch",
+    dockerBaseImage := "openjdk:11-jre-slim",
```
Comment (adaszko) [inline build.sbt]: This may be controversial.  The tests don't pass without it.

**Comment by TomVasile** [comment]
> * Caveats: I'm not sure how Travis will handle uploading build artifacts at the `deploy` stage when there are 2 JDK versions in use.  Does it upload tarballs built with both JDK versions at https://github.com/rchain/rchain/releases/ ?

I would think the files would need to be named differently otherwise they would overwrite.

File: .gitlab-ci.yml (hunk 1 above)
Comment (adaszko) [reply .gitlab-ci.yml]: Done

File: build.sbt (hunk 2 above)
Comment (adaszko) [reply build.sbt]: This entails *always* using openjdk11 in `sbt pushLocal`

**Comment by adaszko** [comment]
> I would think the files would need to be named differently otherwise they would overwrite.

@TomVasile That's my understanding too.  I've added renaming of those files.

File: .gitlab-ci.yml (hunk 1 above)
Comment (ArturGajowy) [inline .gitlab-ci.yml]: we should deduplicate that into an `.sh` script and use it here and above. Similar with other duplication in tasks above. We aren't dropping java8 support anytime soon, it seems to me.

Comment (adaszko) [reply .gitlab-ci.yml]: Makes sense.  I though we were going to ditch 8.

**Comment by ArturGajowy** [comment]
> Why Java 10 is called openjdk-11 in Ubuntu 18.04: lists.ubuntu.com/archives/ubuntu-release/2018-February/004275.html

Please put this into the .yml file itself, or at least in the relevant commit's message. Otherwise it's going to be lost, like tears in the rain...

**Comment by adaszko** [comment]
Superseded by https://github.com/rchain/rchain/pull/1702

**Pull request closed without merging**